src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -772,7 +773,7 @@ void SpatialPooler::updatePermanencesForColumn_(vector<Real>& perm,
   }
 
   numConnected = 0;
-  for (UInt i = 0; i < perm.size(); ++i)
+  for (UInt i = 0; i < perm.size(); i++)
```

Comment:
`i++` and `++i` have no difference in this situation. The former increments the variable `i` and returns the value from before incrementing it. The latter evaluates to the value after incrementing. So the only difference is when you use the result. But the value the statement evaluates to in this for loop is not being used.